File: 167.Two-Sum-II-Input-Array-is-Sorted/pointers.py
class Solution:
    def twoSum(self, numbers: list[int], target: int) -> list[int]:
        i, j = 0, len(numbers) - 1

        # O(n)
        while i < j:
            ss = numbers[i] + numbers[j]

            if ss < target:
                i += 1
            elif ss > target:
                j -= 1
            else:
                return [i + 1, j + 1]

        return []

File: 167.Two-Sum-II-Input-Array-is-Sorted/pointers.rs
use std::collections::HashMap;

impl Solution {
    pub fn two_sum(numbers: Vec<i32>, target: i32) -> Vec<i32> {
        let (mut i, mut j) = (0, numbers.len() - 1);

        // O(n)
        while i < j {
            let sum = numbers[i] + numbers[j];
            if sum < target {
                i += 1;
            } else if sum > target {
                j -= 1;
            } else {
                return vec![1 + i as i32, 1 + j as i32];
            }
        }

        Vec::new()
    }
}
